internal/metrics/metrics.go

```diff
@@ -29,17 +30,27 @@ func RegisterMetrics() {
 	metrics.Registry.MustRegister(containerMaxAllowed)
 }
 
-func RecordContainerRecommendationExcess(vpa *vpav1.VerticalPodAutoscaler) {
+func RecordContainerVpaMetrics(vpa *vpav1.VerticalPodAutoscaler) {
 	// no policy => no maximum => no excess
 	// no recommendations => no excess
 	if vpa.Spec.ResourcePolicy == nil || vpa.Status.Recommendation == nil {
 		return
 	}
+
+	labels := prometheus.Labels{
+		"namespace":             vpa.Namespace,
+		"verticalpodautoscaler": vpa.Name,
+	}
+
 	maxAllowed := make(map[string]corev1.ResourceList)
 	for i := range vpa.Spec.ResourcePolicy.ContainerPolicies {
 		policy := vpa.Spec.ResourcePolicy.ContainerPolicies[i]
 		maxAllowed[policy.ContainerName] = policy.MaxAllowed
+
+		recordMetric(containerMaxAllowed, labels, policy.ContainerName, "cpu", "core", policy.MaxAllowed.Cpu())
+		recordMetric(containerMaxAllowed, labels, policy.ContainerName, "memory", "byte", policy.MaxAllowed.Memory())
 	}
+
```

Comment:
Could you the early return for ` vpa.Status.Recommendation == nil` down here?

Reply:
done